httpstats/transport.go

```diff
@@ -27,22 +29,36 @@ type transport struct {
 	eng       *stats.Engine
 }
 
+// RequestWithContext returns a shallow copy of req with its context changed with this provided tags
+// so the they can be used later during the RoundTrip in the metrics recording.
+// The provided ctx must be non-nil.
+func RequestWithContext(req *http.Request, tags ...Tags) *http.Request {
+	ctx := req.Context()
+	ctx = ctx.WithValue(ctx, contextKeyReqTags, tags)
+	return req.WithContext(ctx)
+}
+
 func (t *transport) RoundTrip(req *http.Request) (res *http.Response, err error) {
 	start := time.Now()
 	rtrip := t.transport
+	eng := t.eng
 
 	if rtrip == nil {
 		rtrip = http.DefaultTransport
 	}
 
+	if tags, ok := req.Context().Value(contextKeyReqTags).([]stats.Tag); ok {
+		eng = eng.WithTags(tags...)
```

Comment:
Wondering if creating a new copy of Engine per request will have a high cpu/memory cost

Reply:
It will add a couple of heap-allocations, but since it's an opt-in feature I think that's fine.